src/indexing/test/indexing.cpp

```diff
@@ -833,16 +833,14 @@ TEST_CASE(
       auto results =
         index_a->get_write_txs_in_range(key, range_start, range_end);
 
-      std::chrono::milliseconds sleep_time(10);
       while (!results.has_value())
       {
         // May be contesting for limited cached buckets with other users of this
-        // index (no handle for unique claims). Back-off exponentially, with
-        // random variation, to break deadlock
+        // index (no handle for unique claims). Uniform random sleep to avoid
+        // deadlock.
+        const auto sleep_time = std::chrono::milliseconds(rand() % 100);
```

Comment:
Question: If it's multi-threaded, and if the expected behavior is for multiple users to be able to share the cached buckets, shouldn't we be handling the deadlock in the code than the test?